src/auth/auth-config.ts

```diff
@@ -1451,3 +1451,68 @@ export class OIDCConfig implements OIDCAuthProviderConfig {
     };
   }
 }
+
+/**
+* Enforcement state of reCAPTCHA protection.
+*   - 'OFF': Unenforced.
+*   - 'AUDIT': Assessment is created but result is not used to enforce.
+*   - 'ENFORCE': Assessment is created and result is used to enforce.
+*/
+export type RecaptchaProviderEnforcementState =  'OFF' | 'AUDIT' | 'ENFORCE';
+
+/**
+* The actions for reCAPTCHA-protected requests.
+*   - 'BLOCK': The reCAPTCHA-protected request will be blocked.
+*/
+export type RecaptchaAction = 'BLOCK';
+
+/**
+* The config for a reCAPTCHA action rule.
+*/
+export interface RecaptchaManagedRule {
+ /**
+  * The action will be enforced if the reCAPTCHA score of a request is larger than endScore.
+  */
+ endScore: number;
+  /**
+  * The action for reCAPTCHA-protected requests.
+  */
+ action?: RecaptchaAction;
+}
+
+/**
+ * The key's platform type: only web supported now..
```

Comment:
nit: extra `.`

Reply:
- Fixed extra `.`.  
- As of the TW review, we were planning to have a TW to do a full review when all pieces in the reCaptcha is ready to be merge into the master branch.
- Yes, the plan is the update the tenant and project config in a separate PR.

Reply:
Sounds good. Thank you!